id_minter/src/main/scala/uk/ac/wellcome/platform/idminter/steps/IdEmbedder.scala

```diff
@@ -2,27 +2,49 @@ package uk.ac.wellcome.platform.idminter.steps
 
 import com.google.inject.Inject
 import com.twitter.inject.Logging
+import io.circe.Json
+import io.circe.optics.JsonPath.root
 import uk.ac.wellcome.metrics.MetricsSender
-import uk.ac.wellcome.models.Work
+import uk.ac.wellcome.models.{SourceIdentifier, Work}
+import io.circe._
+import io.circe.generic.auto._
+import io.circe.parser._
+import io.circe.syntax._
 
 import scala.concurrent.Future
 
 class IdEmbedder @Inject()(metricsSender: MetricsSender,
                            identifierGenerator: IdentifierGenerator)
     extends Logging {
-  def embedId(work: Work): Future[Work] = {
+  def embedId(json: Json): Future[Json] = {
     metricsSender.timeAndCount(
       "generate-id",
       () =>
         Future {
-          val canonicalId = identifierGenerator
-            .retrieveOrGenerateCanonicalId(
-              work.identifiers,
-              ontologyType = work.ontologyType
-            )
-            .get
-          work.copy(canonicalId = Some(canonicalId))
+          val newJson = json.mapObject(addIdentifierToJsonObject)
+          root.items.each.json.modify(addIdentifierToJson)(newJson)
```

Comment:
This could become something like 
```
val keys = List("items", "subjects")

val result = keys.foldLeft(json){
  case (intermediateJson, key) =>
    root.selectDynamic(key).each.json.modify(addIdentifierToObject)(intermediateJson)
}
```
if we have more fields in the root json where we want to add identifiers